attackchains/attackchainhandler.go

```diff
@@ -154,23 +154,41 @@ func (h *AttackChainsEngine) DetectAllAttackChainsFromLists(postureResourceSumma
 	var attackChains []*armotypes.AttackChain
 
 	for i := range postureResourceSummaries {
-		attackTracks, err := h.DetectAllAttackChains(postureResourceSummaries[i], vuls[i])
-		if err != nil {
-			return nil, err
-		}
+		for j := range vuls {
+			if h.workLoadMatch(postureResourceSummaries[i], vuls[j]) {
+				attackTracks, err := h.DetectAllAttackChains(postureResourceSummaries[i], vuls[j])
+				if err != nil {
+					return nil, err
+				}
 
-		if len(attackTracks) == 0 {
-			continue
-		}
+				if len(attackTracks) == 0 {
+					continue
+				}
 
-		currentAttackChains := ConvertAttackTracksToAttackChains(attackTracks, postureResourceSummaries[i])
+				currentAttackChains := ConvertAttackTracksToAttackChains(attackTracks, postureResourceSummaries[i])
 
-		attackChains = append(attackChains, currentAttackChains...)
+				attackChains = append(attackChains, currentAttackChains...)
+			}
+		}
 	}
 
 	return attackChains, nil
 }
 
+func (h *AttackChainsEngine) workLoadMatch(postureResourceSummary *armotypes.PostureResourceSummary, vul *cscanlib.CommonContainerScanSummaryResult) bool {
```

Comment:
So you can remove the existing function "validateWorkLoadMatch"

Reply:
didn't realized we already had this func, I'll use it (I'll just call it earlier)